nuxeo-features/nuxeo-elasticsearch/nuxeo-elasticsearch-audit/src/test/java/org/nuxeo/elasticsearch/TestESHistoryProvider.java

```diff
@@ -191,152 +371,33 @@ protected void createTestEntries() throws Exception {
-        entries = (List<LogEntry>) pp.getCurrentPage();
-
-        log.trace("Verion {} doc history", () -> versions.get(0).getVersionLabel());
-        entries.forEach(entry -> log.trace("LogEntry: {}", entry));
-
-        // creation + 5 updates + update + checkin + created
-        int version1EntriesCount = 1 + 5 + 1 + 1 + 1;
-        if (version1EntriesCount == entries.size()) {
-            assertEquals(Long.valueOf(startIdx).longValue(), entries.get(0).getId());
-            assertEquals(Long.valueOf(startIdx + version1EntriesCount - 1).longValue(),
-                    entries.get(version1EntriesCount - 1).getId());
-        } else {
-            // because update even may be 1ms behind checkin/created !
-            assertEquals(version1EntriesCount - 1, entries.size());
-        }
-
-        // get version 2 history
-        pp = pps.getPageProvider("DOCUMENT_HISTORY_PROVIDER", si, Long.valueOf(20), Long.valueOf(0), new HashMap<>(),
-                versions.get(1));
-        pp.setSearchDocumentModel(searchDoc);
-
-        entries = (List<LogEntry>) pp.getCurrentPage();
-
-        log.trace("Verion {} doc history", () -> versions.get(1).getVersionLabel());
-        entries.forEach(entry -> log.trace("LogEntry: {}", entry));
-
-        // creation + 5x2 updates + checkin/update + checkin + created
-        int versin2EntriesCount = 1 + 5 * 2 + 1 + 1 + 1 + 1;
-        assertEquals(versin2EntriesCount, entries.size());
-        assertEquals(Long.valueOf(startIdx).longValue(), entries.get(0).getId());
-        assertEquals(Long.valueOf(startIdx + versin2EntriesCount).longValue(),
-                entries.get(versin2EntriesCount - 1).getId());
+    protected PageProvider<LogEntry> getPageProvider(String name, int pageSize, int currentPage, Object... parameters) {
+        final List<SortInfo> sorters = List.of(new SortInfo("eventDate", true));
+        @SuppressWarnings("unchecked")
+        PageProvider<LogEntry> pageProvider = (PageProvider<LogEntry>) pageProviderService.getPageProvider(name,
+                sorters, Long.valueOf(pageSize), Long.valueOf(currentPage), Map.of(), parameters);
+        return pageProvider;
+    }
 
+    protected void initSearchDocument(DocumentModel searchDocModel) {
+        updateSearchDocument(searchDocModel, null, null, null, null);
     }
 
+    protected void updateSearchDocument(DocumentModel searchDocModel, List<String> eventsIds,
+            List<String> eventCategories, Date startDate, Date endDate) {
+        searchDocModel.setPropertyValue("basicauditsearch:eventIds",
+                requireNonNullElseGet(eventsIds, List::of).toArray());
+        searchDocModel.setPropertyValue("basicauditsearch:eventCategories",
+                requireNonNullElseGet(eventCategories, List::of).toArray());
+        searchDocModel.setPropertyValue("basicauditsearch:startDate", startDate);
+        searchDocModel.setPropertyValue("basicauditsearch:endDate", endDate);
+    }
```

Comment:
I'm not in favor of this method, it hides what we do to the search document model. For example, I don't know what `updateSearchDocument(doc, List.of(), List.of(eventCateg), null, null)` does. Why do we have null? looks like we handle a corner case everywhere for a specific execution, no?

I'm in favor of having the setPropertyValue lines directly in the test and only the ones we need. Furthermore, re-creating the search doc model each time makes sens to have isolated execution + assertion.